backend/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM node:24
+
+#install application dependencies
+WORKDIR /app
+COPY . .
+
+RUN npm install
+RUN npm run build
```

Comment:
`npm start` defined in https://github.com/patichandana/chippin/blob/main/backend/package.json runs the typescipt compiler and then runs the output js with env file set

1. Do we need always run `tsc` to compile to js before starting the same using node?
2. Start also defines `--env-file` argument, which we already ignored using `.dockerignore`, should this be removed in favour of [docker ENV](https://docs.docker.com/compose/how-tos/environment-variables/set-environment-variables/) so that we can set variables when the container is being run
3. Do we need to copy the entire ts project? or do we just need to copy the build output ( the js files?).